barretenberg/cpp/src/barretenberg/vm2/simulation/lib/db_interfaces.hpp

```diff
@@ -72,14 +72,15 @@ class HighLevelMerkleDBInterface {
   public:
     virtual ~HighLevelMerkleDBInterface() = default;
 
-    virtual const TreeSnapshots& get_tree_roots() const = 0;
+    virtual InternalTreeSnapshots get_tree_roots() const = 0;
```

Comment:
Same comment as before. Marking it here because it's the interface. We can resolve all of them at the same time if/when we find a solution.